staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -63,6 +63,9 @@ func WithAuthentication(handler http.Handler, mapper genericapirequest.RequestCo
 				if err != nil {
 					glog.Errorf("Unable to authenticate the request due to an error: %v", err)
 				}
+				if auditFailedAuthn != nil {
+					w = auditFailedAuthn.Decorate(w, user, req)
```

Comment:
why are you passing user? when `err != nil || !ok`, `user` is unusable

Reply:
and if you don't need user, then `auditFailedAuthn.Decorate` has the same args as `http.Handler`, and you can simple pass a wrapped failed http.Handler in here that logs, then handles